R/utils.R

```diff
@@ -161,3 +161,33 @@ df_has_base_subset <- function(x) {
   method <- .Call(vctrs_s3_find_method, "[", x, table)
   is_null(method) || identical(method, `[.data.frame`)
 }
+
+last <- function(x) {
+  x[[length(x)]]
+}
+
+vec_common_suffix <- function(x, y) {
```

Comment:
A comment about how this works might be useful, I spent a few minutes reading it to be able to figure it out.